### PR TITLE
[6_1_0][TIMOB-23429] Android: Nougat file schematics

### PR DESCRIPTION
**Pull request by frankieandone**
#### Test case:

Should not get `FileUriExposedException` error.

#### app.js (set up camera):
```
var win;
function fireCamera() {
	if (Ti.Platform.osname === 'android' || Ti.Platform.osname == "iphone" || Ti.Platform.osname == 'ipad') {
		win.removeEventListener('focus', fireCamera);
	}
	Titanium.Media.showCamera({

		success: function (event) {
			var cropRect = event.cropRect;
			var image = event.media;

			if (event.mediaType == Ti.Media.MEDIA_TYPE_PHOTO) {
				var imageView = Ti.UI.createImageView({
					width: win.width,
					height: win.height,
					image: event.media
				});
				win.add(imageView);
			}
		},
		cancel: function () {

		},
		error: function (error) {
			if (error.code == Titanium.Media.NO_CAMERA) {
				console.log("device has no camera", error);
			}
			else {
				console.log("unexpected error", error);
			}
		},
		saveToPhotoGallery: true,
		allowEditing: true,
		mediaTypes: [Ti.Media.MEDIA_TYPE_PHOTO]
	});
}
win = Titanium.UI.createWindow({
	title: "classic_app"
});
win.open();
if (Ti.Platform.osname === 'android' || Ti.Platform.osname == "iphone" || Ti.Platform.osname == 'ipad') {
	win.addEventListener('focus', fireCamera);
} else {
	fireCamera();
}
```
#### AndroidManifest (add permissions and target Android 7):
```
<?xml version="1.0" encoding="UTF-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android">
	<uses-sdk android:minSdkVersion="14" android:targetSdkVersion="25"/>
        ....
	<uses-permission android:name="android.permission.CAMERA"/>
	<uses-feature android:name="android.hardware.camera2"/>
</manifest>
```

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23429

### CODE REVIEW

**Comment by hansemannn** [comment]
@fmerzadyan No reviewer assigned here. Assigning @garymathews as per ticket.

**Comment by garymathews** [comment]
Please refer to the changes request in the main PR here: https://github.com/appcelerator/titanium_mobile/pull/8706#pullrequestreview-38954085

**Review by garymathews — CHANGES_REQUESTED** [review]
Please refer to the changes request in the main PR here: https://github.com/appcelerator/titanium_mobile/pull/8706#pullrequestreview-38954085

**Comment by fmerzadyan** [comment]
@garymathews, added requested changes.

**Review by garymathews — APPROVED** [review]
CR: **PASS**

##### TEST CASE
```JS
var win = Titanium.UI.createWindow({title: 'TIMOB-23429', backgroundColor: 'gray', layout: 'vertical'}),
    btn = Ti.UI.createButton({title: 'TAKE PICTURE', width: '100%', height: '40dp'}),
    iv = Ti.UI.createImageView({autorotate: true, width: Ti.UI.FILL, height: Ti.UI.FILL});

function takePicture() {
    Titanium.Media.showCamera({
        success: function(e) {
            iv.image = e.media;
            Ti.Media.hideCamera();
        },
        cancel: function() {
            Ti.API.info('canceled');
        },
        error: function(e) {
            Ti.API.error(JSON.stringify(e, null, ' '))
        },
        saveToPhotoGallery: true,
        mediaTypes: [Ti.Media.MEDIA_TYPE_PHOTO]
    });
}

btn.addEventListener('click', function(e) {
    var permissions = ['android.permission.CAMERA',
                       'android.permission.WRITE_EXTERNAL_STORAGE',
                       'android.permission.READ_EXTERNAL_STORAGE'];
    for (var permission of permissions) {
        if (!Ti.Android.hasPermission(permission)) {
            Ti.Android.requestPermissions(permissions, function(e) {
                if (e.success) {
                    takePicture();
                }
            });
            return;
        }
    }
    takePicture();
});

win.add(btn);
win.add(iv);
win.open();
```

File: android/cli/commands/_build.js [hunk 1]
```diff
@@ -3408,6 +3409,22 @@ AndroidBuilder.prototype.generateI18N = function generateI18N(next) {
 	next();
 };
 
+AndroidBuilder.prototype.generateFileProviderPaths = function generateFileProviderPaths(next) {
+	var xmlDir = path.join(this.buildResDir, 'xml');
+
+	if fppFile = path.join(xmlDir, 'file_provider_paths.xml');
```
Comment (lokeshchdhry) [inline android/cli/commands/_build.js]: `if fppFile` should be 'var fppFile`. Build fails due to this.

**Review by jquick-axway — APPROVED** [review]
CR: Pass

**Comment by lokeshchdhry** [comment]
FR Passed.

The implementation works as expected according to the new permission model for android N.

Studio Ver:     4.9.0.201705170123
SDK Ver:        6.1.0 Local build
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.2
Appc NPM:       4.2.9
Appc CLI:       6.2.1
Ti CLI Ver:     5.0.13
Alloy Ver:      1.9.11
Node Ver:       6.10.1
Java Ver:       1.8.0_101
Devices:        ⇨ google Nexus 6 --- Android 6.0.1
                ⇨ google Pixel XL --- Android 7.1.1

**Pull request merged**